### PR TITLE
Adds modifier class to differentiate share and follow

### PR DESCRIPTION
**Pull request by anselmbradford**
Adds modifier class to differentiate share and follow social media icons as requested by the analytics team.
Updates careers, blog, and events where old sharing template was used.

## Changes

- Updates the social media molecule to `display: inline-block` so that it doesn't have right padding.
- Adds `m-social-media__share` and `m-social-media__follow` modifiers for aiding analytics.
- Fixes issue where students-and-graduates was set to hide media images on mobile, but wasn't.

## Testing

- `gulp build`
- Footer share icons should have `m-social-media__follow` class.
- `/about-us/the-bureau/about-director/` share icons should have `m-social-media__share` class.
- `/blog/`, `/careers/`, `/about-us/events/` should have Share This icons.

## Review

- @bethannenc
- @kelleyholden
- @KimberlyMunoz 
- @sebworks 
- @schaferjh 

## Screenshots

/blog/
screen shot 2016-03-31 at 1 32 19 pm

----

/about-us/events/[detail]

screen shot 2016-03-31 at 1 47 26 pm

mobile:
screen shot 2016-03-31 at 1 47 34 pm

----

/careers/

screen shot 2016-03-31 at 1 54 41 pm

Fix for hide on mobile on /careers/students-and-graduates/
screen shot 2016-03-31 at 1 28 17 pm

## Todos

- video share needs to be updated to use the atomic social media molecule and then the old share macro can be tossed.

### CODE REVIEW

File: cfgov/unprocessed/css/pages/careers.less [hunk 1]
```diff
@@ -3,10 +3,22 @@
    Career pages at /careers/.
    ========================================================================== */
 
-// TODO: Consolidate site-wide media_image responsive sizes into one class.
-.students-and-graduates .media_image {
-    width: 130px;
-    .respond-to-min(@bp-med-min, {
-        width: 150px;
-    });
+.t-careers {
```
Comment (KimberlyMunoz) [inline cfgov/unprocessed/css/pages/careers.less]: What does the .t modifier stand for?

**Comment by KimberlyMunoz** [comment]
:+1:

File: cfgov/unprocessed/css/pages/careers.less (hunk 1 above)
Comment (anselmbradford) [inline cfgov/unprocessed/css/pages/careers.less]: Sorry, I should have noted. Template. I figured we do that with organisms, molecules, atoms.

File: cfgov/jinja2/v1/_includes/article.html [hunk 2]
```diff
@@ -79,7 +79,7 @@ <h1 class="post_heading">
                 {{ time.render(post.date, {'date':true}) }}
             </span>
         </div>
-        {{ share.render({
+        {{ social_media.render({
             'title': post.title,
             'hide_heading': true,
```
Comment (anselmbradford) [inline cfgov/jinja2/v1/_includes/article.html]: Note to self: remove L:83-84

**Pull request merged**